editor/src/components/inspector/common/longhand-shorthand-hooks.ts

```diff
@@ -245,14 +231,92 @@ export function useInspectorInfoLonghandShorthand<
       controlStyles: getControlStyles(controlStatus),
       onSubmitValue: onSubmitValue,
       onTransientSubmitValue: onTransientSubmitValue,
-      onUnsetValues: onUnsetValues,
+      onUnsetValues: NO_OP,
       orderedPropKeys: orderedPropKeys,
     }
   })
 
-  return mapArrayToDictionary(
+  const longhandResultsDictionary = mapArrayToDictionary(
     longhandResults,
     (_, index) => longhands[index],
     (result) => result,
   )
+
+  const longhandResultsWithUnset = objectMap((longhandResult, longhandToUnset) => {
+    const onUnsetValues = () => {
+      const allPropKeysEqual = allOrderedPropKeys.every((propKeys) => {
+        return arrayEquals(propKeys, allOrderedPropKeys[0])
+      })
+      if (!allPropKeysEqual) {
+        // we do nothing for now. we cannot ensure that we can make a sensible update and surface it on the UI as well
+        return
+      }
+      const propkeysToUse = allOrderedPropKeys[0]
+
+      const actionsToDispatch = (() => {
```

Comment:
Does this really need to be an immediately invoked anonymous function?

Reply:
I'm testing the water with this one. when we were refactoring the `useInspectorInfo` hook, it was made harder by the pattern we use when we make a mutable variable at the top of the function, and then assign a value to it in various conditional branches. I have two problems: the mutable variable remains mutable even after the conditions because unfortunately typescript doesn't have an "assign once" const, the other problem is that you can't `break` or `return` in a condition so you cannot guarantee someone accidentally overwriting your value later down the line.

we talked about using an immediately executed function to simulate the missing javascript feature of `if`s being expressions.

Reply:
I'd be tempted to just make it an inner function and return/use the value after invoking it.

Reply:
I turned it into a regular function in the module scope, it was on the verge of being too big to be inline